src/App.tsx

```diff
@@ -8,37 +8,203 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getAllUsers } from './api/users';
+import { User } from './types/User';
+import { getUserPosts } from './api/posts';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
+import { getPostComments, addNewComment, deleteComment } from './api/comments';
+
+const tempUser = {
+  id: 0,
+  name: '',
+  email: '',
+  phone: '',
+};
+
+const tempPost = {
+  id: 0,
+  userId: 0,
+  title: '',
+  body: '',
+};
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>(tempUser);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [isPostLoading, setIsPostLoading] = useState(false);
+  const [isPostLoadingError, setIsPostLoadingError] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [openedPost, setOpenedPost] = useState<Post>(tempPost);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+  const [isCommentLoadingError, setIsCommentLoadingError] = useState(false);
+  const [isNoComments, setIsNoComments] = useState(false);
+  const [isNewCommentForm, setIsNewCommentForm] = useState(false);
+
+  const loadUsers = async () => {
+    const loadedUsers = await getAllUsers();
+
+    if (loadedUsers) {
+      setUsers(loadedUsers);
+    }
+  };
+
+  const loadUserPosts = async (userId: number) => {
+    setUserPosts([]);
+
+    try {
+      const loadedUserPosts = await getUserPosts(userId);
+
+      if (loadedUserPosts.length !== 0) {
+        setIsPostLoading(false);
+        setUserPosts(loadedUserPosts);
+      } else {
+        setIsPostLoading(false);
+        setIsNoPosts(true);
+      }
+    } catch {
```

Comment:
Use a finally block to `setIsPostLoading(false) ` to avoid duplicating the code in both try and catch blocks.